client/asset/btc/btc.go

```diff
@@ -648,6 +648,12 @@ func (btc *ExchangeWallet) feeRate(confTarget uint64) (uint64, error) {
 	return 1 + uint64(satPerKB)/1000, nil
 }
 
+type sat uint64
+
+func (s sat) String() string {
+	return strconv.FormatFloat(btcutil.Amount(s).ToBTC(), 'f', -1, 64) // dec, but no trailing zeros
+}
```

Comment:
I'm all for a custom type for this purpose, but I was a little confused since the name is `sat`, but it's primary purpose is for a string in conventional units. Confusion isn't helped by the way it is used, `sat(v)`, which could appear to be a function call to convert a conventional unit value to Satoshis. 

Since it's handling values in two representations, how about the more ambiguous `type amount uint64`?

Reply:
I think it just needs to be viewed as a compatible type conversion.  That is, you assert that the value you have is Satoshis via `sat(v)`.  This is similar to doing `dcrutil.Amount(someInt64)`, which gives you an `Amount` with the same underlying type, just new methods.  Admittedly it just has one method, so the value of Stringerization vs a function call is limited.  I will go back and toy with a function, but I think making it more ambiguous with `amount` vs. `sat` is actually more confusing.

Part of the consideration for this approach (stringer vs function) is that in a log call, it will only call the stringer when the debug level requires evaluating the stringer, but that's admittedly not a strong reason.